### PR TITLE
feat: add oracle upgrade related genesis

### LINKED ISSUE #581: Implement OracleUpgrade and OracleUpgradeInfo to genesis

### PR DESCRIPTION
**Pull request by audtlr24**
closes #581 .

### CODE REVIEW

**Review by H4NLee — APPROVED** [review]
lgtm

File: x/oracle/types/genesis.go [hunk 1]
```diff
@@ -41,6 +43,18 @@ func (m *GenesisState) Validate() error {
 		}
 	}
 
+	for _, oracleUpgrade := range m.OracleUpgrades {
+		if err := oracleUpgrade.ValidateBasic(); err != nil {
+			return err
+		}
+	}
+
+	if m.OracleUpgradeInfo != nil {
```
Comment (inchori) [inline x/oracle/types/genesis.go:52]: Does the `OracleUpgradeInfo` in `GenesisState` should be always not nil?

**Review by inchori — APPROVED** [review]
Gooood!
Just one question.

File: x/oracle/types/genesis.go (hunk 1 above)
Comment (inchori) [reply x/oracle/types/genesis.go:52]: Ah.. I understand it. Please forgot this comment.

Comment (gyuguen) [reply x/oracle/types/genesis.go:52]: If we follow the specs in this [link](https://github.com/medibloc/panacea-core/pull/578#discussion_r1059859889), I think we should check it like @inchori said.

**Review by gyuguen — APPROVED** [review]
LGTM

File: x/oracle/types/genesis.go (hunk 1 above)
Comment (H4NLee) [reply x/oracle/types/genesis.go:52]: right!

Comment (H4NLee) [reply x/oracle/types/genesis.go:52]: The `OracleUpgradeInfo` can be nil if there is no upgrade.
It seems no problem.

Comment (gyuguen) [reply x/oracle/types/genesis.go:52]: Aha, I was confused about how far the upgrade action could go.
If the upgrade command cannot be performed as long as the upgrade action does not occur, now is OK.
However, even if there is no upgrade operation, if the upgrade command is available (i.e. OracleShareKey is lost), we must set uniqueID in params and upgradeInfo when launching DEP.

As @H4NLee said before, if oracle can't use this address when the key is lost, no problem.

**Pull request merged**